Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -2347,7 +2347,41 @@ private class GenericArrayElementLvalue : ILvalue {
       var psi = PrepareProcessStartInfo("javac", new List<string> { "-encoding", "UTF8" }.Concat(files));
       psi.WorkingDirectory = Path.GetFullPath(Path.GetDirectoryName(targetFilename));
       psi.EnvironmentVariables["CLASSPATH"] = GetClassPath(targetFilename);
-      return 0 == RunProcess(psi, outputWriter, "Error while compiling Java files.");
+      bool compileIsSuccess = 0 == RunProcess(psi, outputWriter, "Error while compiling Java files.");
+      if (compileIsSuccess) {
+        if (!DafnyOptions.O.UseRuntimeLib) {
+          // If the default runtime lib was not used for compiling, then we should not include it
+          // in the jar either.
+          // SHould we have an option to exclude it anyway if we are making a library jar
+          psi = PrepareProcessStartInfo("jar", new List<String> { "xf", "DafnyRuntime.jar" });
+          psi.WorkingDirectory = Path.GetFullPath(Path.GetDirectoryName(targetFilename));
+          compileIsSuccess = 0 == RunProcess(psi, outputWriter, "Error while creating jar file (unzipping runtime library).");
+        }
+
+        var classfiles = new List<String>();
+        foreach (string file in Directory.EnumerateFiles(targetDirectory, "*.class", SearchOption.AllDirectories)) {
+          classfiles.Add(Path.GetRelativePath(targetDirectory, file));
+        }
+        string simpleName = Path.GetFileNameWithoutExtension(targetFilename);
+        // Put the jar in the current working directory
+        string jarPath = Path.GetFullPath("./" + simpleName + ".jar");
```

Comment:
That would be fine with me, but it would be a breaking change. The problem is that in the Java compiler, with `dafny build`, with this PR, there is no longer any need for the built artifacts other than the .jar (if there is, the user can keep them with --spill-target-code). But before this PR, users (and tests) would rely on the location of the output directory to run the compiled Java code.

Now, we could make the change that with the new CLI, the location is different, but leave it the same for the old CLI.

Reply:
Is there a way for the user in `dafny build` to specify other Java dependencies?
If I have Dafny externs and I include them does it just work?

Reply:
1) The jar file is now placed in --output; the generated source was never in --output but in a directory with the name of --output and "-java" appended. So we leave that alone so as not to break anything and it serves as a temporary directory.

2) Yes, other .java files can be placed on the command lie and they "just work". If the program depends on other jar files, they have to be on the command-line, both for compiling/building and running.